luminaire/model/window_density.py

```diff
@@ -545,23 +549,47 @@ def _get_result(self, input_df=None, detrend_order=None, agg_data_model=None, va
             execution_data_avg = np.mean(execution_data)
             # If detrending is needed, we scale the scoring data accordingly using the agg_dat_model forecast
             if detrend_order > 0:
+                snapshot_len_max = min(len(agg_data), len_req_agg_data_model)
+                agg_data_trunc = np.array(agg_data)[:, 1][-snapshot_len_max:]
+                data_adjust_forecast = []
                 try:
-                    data_adjust_forecast = agg_data_model.score(execution_data_avg, scoring_datetime)['Prediction'] \
-                        if agg_data_model else agg_data[-1][-1]
+                    if agg_data_model and len(agg_data) > len_req_agg_data_model:
+                        score = agg_data_model.score(execution_data_avg, scoring_datetime)
+                        data_adjust_forecast.append(score['Prediction'])
+                        data_adjust_forecast.append(score['CILower'])
+                        data_adjust_forecast.append(score['CIUpper'])
+                    else:
+                        data_adjust_forecast.append(np.median(agg_data_trunc))
+                        data_adjust_forecast.append(np.percentile(agg_data_trunc, 5))       # setting a 2-sigma limit
+                        data_adjust_forecast.append(np.percentile(agg_data_trunc, 95))       # setting a 2-sigma limit
                 except:
                     # If the scoring for the agg_data_model fails for some reason, we use the latest agg_data for the
                     # detrending adjustment
-                    data_adjust_forecast = agg_data[-1][-1]
-                adjusted_execution_data = (execution_data / data_adjust_forecast).tolist()
+                    data_adjust_forecast.append(np.median(agg_data_trunc))
+                    data_adjust_forecast.append(np.percentile(agg_data_trunc, 5))       # setting a 2-sigma limit
+                    data_adjust_forecast.append(np.percentile(agg_data_trunc, 95))       # setting a 2-sigma limit
+                for i in range(3):
+                    if data_adjust_forecast[i] != 0:
+                        adjusted_execution_data.append((execution_data / data_adjust_forecast[i]).tolist())
             else:
                 adjusted_execution_data = list(execution_data)
 
         # Kl divergence based anomaly detection
         if detection_method == "kldiv":
-            current_anomaly_score = self._distance_function(data=adjusted_execution_data,
-                                                            called_for="scoring", baseline=baseline)
-            prob_of_anomaly = st.gamma.cdf(current_anomaly_score, anomaly_scores_gamma_alpha,
-                                           anomaly_scores_gamma_loc, anomaly_scores_gamma_beta)
+            if detrend_order > 0:
```

Comment:
Can we add some comments on why we are checking >0 only? The only def of variable `detrend_order` I see is ```:param int detrend_order: The order of detrending based on MA or differencing method.``` So, not sure what range of values are expected here and what it means when is a positive, zero or negative number.